pkgs/native_toolchain_c/test/clinker/treeshake_test.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+//TODO(mosuem): Enable for windows and mac.
+@TestOn('linux')
+library;
+
+import 'dart:io';
+
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+import 'package:test/test.dart';
+
+import '../helpers.dart';
+import 'build_testfiles.dart';
+
+Future<void> main() async {
+  if (!Platform.isLinux) {
+    // Avoid needing status files on Dart SDK CI.
+    return;
+  }
+
+  CLinker linkerManual(List<String> sources) => CLinker.library(
+        name: 'mylibname',
+        assetName: '',
+        sources: sources,
+        linkerOptions: LinkerOptions.manual(
+          flags: ['--strip-debug', '-u', 'my_other_func'],
+          gcSections: true,
+          linkerScript: Uri.file('test/clinker/test_data/linker/symbols.lds'),
+        ),
+      );
+  CLinker linkerAuto(List<String> sources) => CLinker.library(
+        name: 'mylibname',
+        assetName: '',
+        sources: sources,
+        linkerOptions: LinkerOptions.treeshake(symbols: ['my_other_func']),
+      );
+  CLinker linkerAutoEmpty(List<String> sources) => CLinker.library(
+        name: 'mylibname',
+        assetName: '',
+        sources: sources,
+        linkerOptions: LinkerOptions.treeshake(symbols: null),
+      );
+  const architecture = Architecture.x64;
+  const os = OS.linux;
+
+  late Map<String, int> sizes;
+  sizes = <String, int>{};
+  for (final clinker in [
+    (name: 'manual', linker: linkerManual),
+    (name: 'auto', linker: linkerAuto),
+    (name: 'autoEmpty', linker: linkerAutoEmpty),
+  ]) {
+    test('link test with CLinker ${clinker.name}', () async {
```

Comment:
Please add a loop over all target architectures around this test.
(Or alternatively, add another `UnsupportedError` for of target arch != host arch and untick the checkmark on the tracking issue that Linux is supported.)

And check that the outputted dylib is for the right architecture.
See pkgs/native_toolchain_c/test/cbuilder/cbuilder_cross_linux_host_test.dart on how to do that.

Reply:
Why was this marked as resolved? I don't see test coverage or unsupported errors? (Do you have pending commits?)

Reply:
I do use that feature of the VS code extension sometimes, but the Web UI doesn't have it unfortunately.